### ISSUE TITLE
A/B Street is no longer loading

### ISSUE THREAD

**Issue by mvl22** [cyipt/actdev-ui#81]
The simulation is no longer loading:
https://actdev.cyipt.bike/abstreet/?--actdev=allerton-bywater&--cam=12.0/53.74523/-1.36122&--actdev_scenario=base

as linked from
https://actdev.cyipt.bike/allerton-bywater/

The error is: "Error Loading App: unable to fetch wasm - contentLength was unexpectedly undefined"

**Comment by mvl22** [comment]
@dabreegster Any ideas what might be going wrong? Is this likely to be an easy fix?

**Comment by Robinlovelace** [comment]
This has been the case for 12+ months I think. Good on you for reporting it Martin, is very cool for live demos and I hope it can live on.

**Comment by dabreegster** [comment]
Huh, weird. Everything should be totally self-contained, so unless the hosted files changed, not sure why this would start breaking. The error message is not enlightening.

We could pull in the latest WASM build + files from that release, check they work, and host those instead. But I stopped maintaining the traffic sim in the later [releases](https://github.com/a-b-street/abstreet/releases), so more things probably break. Let me find the best release to use and ping here with the URLs of all the files to go mirror. Will try to get to it this weekend, but guessing this is not urgent, especially if it's been broken for a whole year? :(

**Comment by dabreegster** [comment]
> contentLength was unexpectedly undefined

Ah wait, this makes me think it might be the JS frontend logic that draws a progress bar for fetching the WASM blob. Did your server settings change in the last year to stop sending the `Content-Length` header maybe? But `curl -v https://actdev.cyipt.bike/abstreet/abstreet/wasm_pkg/game_bg.wasm` seems to show it.

**Comment by mvl22** [comment]
> not urgent

Not urgent, though I'd like to get this going again.

I was auditing what's currently on our server before we redeploy it to a new one.

**Comment by mvl22** [comment]
Now fixed in c630ba67fb47a1642809f743047e70c75df6c040.

Seems like gzip encoding on the server was causing the Content-Length not to be sent, as that switches large files into chunked mode. I've force-disabled Gzip serving now.

**Comment by Robinlovelace** [comment]
Fantastic, thank you Martin!

**Comment by mvl22** [comment]
> Ah wait, this makes me think it might be the JS frontend logic that draws a progress bar for fetching the WASM blob.

@dabreegster Yes, it probably was that. That logic should probably be tolerant of a server not having a Content-Length, since there could well be chunking on a large file.

**Comment by Robinlovelace** [comment]
Fix confirmed. Tested on the Chapeltown example and currently simulating myself!

![Image](https://github.com/user-attachments/assets/29093405-9bff-4388-bcc3-1f1d356666a2)

**Comment by Robinlovelace** [comment]
https://actdev.cyipt.bike/chapeltown-cohousing/routenetwork,buildings/#12/53.81071/-1.52711

**Comment by Robinlovelace** [comment]
https://actdev.cyipt.bike/abstreet/?--actdev=chapeltown-cohousing&--cam=15.71/53.80868/-1.52822&--actdev_scenario=base

**Comment by dabreegster** [comment]
Thanks for fixing this! This issue was buried lowest on my TODO list. Noted about being resilient to being served from different places; originally we wrote it assuming we'd control all deployments.
